include/swift/Reflection/ReflectionContext.h

```diff
@@ -415,24 +428,32 @@ class ReflectionContext
 
     assert(SecIdx < SecHdrVec.size() && "malformed ELF object");
 
-    const typename T::Section *SecHdrStrTab = SecHdrVec[SecIdx];
+    std::unique_ptr<const typename T::Section> &SecHdrStrTab =
+        SecHdrVec[SecIdx];
     typename T::Offset StrTabOffset = SecHdrStrTab->sh_offset;
     typename T::Size StrTabSize = SecHdrStrTab->sh_size;
 
-    auto StrTabStart =
-        RemoteAddress(ImageStart.getAddressData() + StrTabOffset);
-    auto StrTabBuf = this->getReader().readBytes(StrTabStart, StrTabSize);
-    auto StrTab = reinterpret_cast<const char *>(StrTabBuf.get());
+    // The string table is also not mapped in the child process,
+    // read its contents from the file buffer.
+    uint64_t StrTabStart = FileBufferStartAddress + StrTabOffset;
 
     auto findELFSectionByName = [&](std::string Name)
         -> std::pair<RemoteRef<void>, uint64_t> {
       // Now for all the sections, find their name.
-      for (const typename T::Section *Hdr : SecHdrVec) {
+      for (auto &&Hdr : SecHdrVec) {
         uint32_t Offset = Hdr->sh_name;
-        auto SecName = std::string(StrTab + Offset);
+        char *Start = (char *)StrTabStart + Offset;
+        uint64_t Size = 0;
+        while (Start[Size] != '\0' && Size <= StrTabSize - Offset)
+          ++Size;
```

Comment:
let's use `strnlen` or `strnlen_s` for this. It's going to be faster and less likely to have a buggy implementation.